### ISSUE TITLE
JavadocParagraph should complain about missing blank line before at-clauses

### ISSUE THREAD

**Issue by josephmate** [checkstyle/checkstyle#6192]
https://checkstyle.org/config_javadoc.html#JavadocParagraph

```
side_projects / checkStyleIssue / $ javac Test.java
side_projects / checkStyleIssue / $ cat Test.java
public class Test {

    /**
     * Checkstyle should complain about this line or the following line.
     * @param param This param does something
     */
    public void function(String param) {
    }

}
side_projects / checkStyleIssue / $ cat checkstyle.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://www.puppycrawl.com/dtds/configuration_1_3.dtd">

<module name="Checker">
    <!-- Main class walker -->
    <module name="TreeWalker">
        <module name="JavadocParagraph"/>
    </module>
</module>


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar ~/Downloads/checkstyle-8.14-all.jar -c checkstyle.xml Test.java 
Starting audit...
Audit done.
```

https://checkstyle.org/config_javadoc.html#JavadocParagraph
I expect checkstyle to complain about line 4, or 5 because JavadocParagraph docs say:

>  Checks that ... There is ... one blank line before the at-clauses block if it is present. 

The quotation is taken from http://checkstyle.sourceforge.net/config_javadoc.html#JavadocParagraph

**Comment by rnveach** [comment]
Going back to the original implementation at https://github.com/checkstyle/checkstyle/issues/315 , I don't think this functionality was ever added as I couldn't get violations on older versions.

**Comment by romani** [comment]
This check was created to cover https://checkstyle.org/google_style.html#a7.1.2 , weird that we missed this requirement.

In need to be investigated on to fix the Check or create new Check for empty line separation

**Comment by ashleyheath** [comment]
@romani I'd be interested in fixing this - was it ever decided whether the fix belonged in the existing check or if it belonged in a new one?

**Comment by pbludov** [comment]
I would like also a check for redundant blank lines such as 
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/WrapOption.java#L24
However, I'm not sure it's related to this issue. Just a thought.

**Comment by romani** [comment]
From Check documentation:
>  Checks that:
    There is one blank line between each of two paragraphs and one blank line before the at-clauses block if it is present. 

we need to fix this Check, it might be some bug present.

**Comment by josephmate** [comment]
Ok. I think I have a fix coming along here https://github.com/josephmate/checkstyle/commit/1ce30507e35f39debf29691e011ba29ded69dd85

but it needs debugging.

**Comment by romani** [comment]
@josephmate , there will be violations in our code when this issue is resolved. We need to update our code to resolve violation, please do this as separate commit (do not fix such changes with functional fix). Just follow `mvn verify`  and CI .... it tell you what you have to do.

**Comment by josephmate** [comment]
@romani : 1 PR with two commits, or 2 PRs, each with one commit?

**Comment by romani** [comment]
Either is ok

**Comment by romani** [comment]
https://google.github.io/styleguide/javaguide.html#s7.1.2-javadoc-paragraphs
> One blank line—that is, a line containing only the aligned leading asterisk (*)—appears between paragraphs, and before the group of block tags if present.

This Check was created based on Google style requirements. Few years are there was that term "at-clause" ([patched at](https://github.com/google/styleguide/commit/594d91bfbb8dfcbcac0f148af8175f2b7ec9857b), [issue from us on this](https://github.com/google/styleguide/issues/215)). Fortunately no such term any more, there is `group of block tags` so name of Check should contain "BlockTagsGroup".

we need to remove this term from https://checkstyle.org/config_javadoc.html#JavadocParagraph also, to avoid any further confusion.

**Comment by romani** [comment]
@josephmate , thanks a lot for hard work to make it done. Sorry for long process.

Fix is merged
